server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -522,150 +529,229 @@ public IndexMetaDataGenerations indexMetaDataGenerations() {
      *                           from cached bytes that we trust to not contain broken generations.
      */
     public static RepositoryData snapshotsFromXContent(XContentParser parser, long genId, boolean fixBrokenShardGens) throws IOException {
+        if (parser.nextToken() != XContentParser.Token.START_OBJECT) {
```

Comment:
We have utility methods in `XContentParserUtils` for this:

```suggestion
ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
```

Reply:
I know, I actually wanted to use it here in all these spots but then I realized that it's pretty inefficient in it's current form. I fixed it to behave better in https://github.com/elastic/elasticsearch/pull/62691 and would rather wait for that PR to go in before making that change to not make this method even larger (in terms of what it compiles to).

Reply:
Right - I did not look at all the ctors, thanks